### PR TITLE
Resurrecting KubernetesPipelineTest.runInPodWithDifferentShell

### PR DESCRIPTION
**Pull request by jglick**
Reverting #771 because it turns out that this was actually testing something useful: that the build fails promptly if you specify the wrong `shell` for a `container` step. The behavior as of then was not great (`IOException: Pipe closed` rather than the more helpful `no such file or directory` it emitted at one point) but after #757 it is much worse: the `sh` step simply hangs. CC @ccojocar

### CODE REVIEW

**Comment by Vlatombe** [comment]
@jglick Not sully satisfied with my solution because it adds a delay on success case. Let me know what you think

**Review by jglick — COMMENTED** [review]
Probably do not understand the implementation here without going through in a debugger, but it sounds OK. Really we should be testing two cases:

* A regular `Launcher`-driven synchronous nonblocking build step (e.g. `SimpleBuildStep`).
* A `sh` step, presuming the `durable-task` dep is recent.
* Same but with `FlagRule` turning on `DurableTaskStep.USE_WATCHING`.

File: src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java [hunk 1]
```diff
@@ -95,6 +95,7 @@ public void setUp() throws Exception {
         deletePods(cloud.connect(), getLabels(cloud, this, name), false);
         warnings.record("", Level.WARNING).capture(1000);
         assertNotNull(createJobThenScheduleRun());
+        logs = logs.recordPackage(ContainerExecDecorator.class, Level.FINEST);
```
[the remark this answers is not in the document]
Comment (Vlatombe) [reply src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java:98]: Ah, need to revert this line, It becomes too verbose for what Jenkins collects. Was useful to debug though.

Comment (Vlatombe) [inline src/test/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesPipelineTest.java:98]:
```suggestion
```

**Comment by Vlatombe** [comment]
The problem here is that if the shell does not exist in the container, the `kubectl exec` immediately fails, however the only time where this is actually checked is in https://github.com/jenkinsci/kubernetes-plugin/blob/f6590b36b269f99f0ef9add3950161bafdb960fc/src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecProc.java#L85, which seems to be never called when using durable task.

Since the rest of durable-task assumes everything (result, log, etc.) is obtained via temporary files, and these are never written since the holder script is never run, then it waits forever.

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java [hunk 2]
```diff
@@ -547,7 +547,7 @@ public void close() throws IOException {
             try {
                 closable.close();
             } catch (Exception e) {
-                LOGGER.log(Level.FINE, "failed to close {0}");
+                LOGGER.log(Level.FINE, "failed to close {0}")   ;
```
Comment (jglick) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java:550]:
```suggestion
                LOGGER.log(Level.FINE, "failed to close", e);
```

??

**Comment by Vlatombe** [comment]
Not sure why, but the test result in CI differs from my local machine where I get the expected output.

```
[run In Pod With Different Shell #1] ERROR: Process exited immediately after creation. See output below
[run In Pod With Different Shell #1] Executing /bin/bash script inside container busybox of pod run-in-pod-with-different-shell-1-t18hc-jsrv7-6bg2d
[run In Pod With Different Shell #1] OCI runtime exec failed: exec failed: container_linux.go:370: starting container process caused: exec: "/bin/bash": stat /bin/bash: no such file or directory: unknown
```

**Pull request merged**